src/core/src/op/select.cpp

```diff
@@ -45,120 +74,72 @@ void op::v1::Select::validate_and_infer_types() {
-                     const op::AutoBroadcastSpec& autob,
-                     const element::Type_t& et) {
-    bool rc = false;
-
-    switch (et) {
-        OPENVINO_TYPE_CASE(evaluate_select, i8, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, i16, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, i32, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, i64, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, u8, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, u16, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, u32, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, u64, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, bf16, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, f16, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, f32, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, f64, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, boolean, output_values, input_values, autob);
-    default:
-        rc = false;
-        break;
-    }
-
-    return rc;
-}
-}  // namespace
-}  // namespace detail
-
-bool op::v1::Select::evaluate(const HostTensorVector& output_values, const HostTensorVector& input_values) const {
+bool Select::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v1_Select_evaluate);
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    OPENVINO_ASSERT(validate_host_tensor_vector(input_values, 3));
-    OPENVINO_ASSERT(validate_host_tensor_vector(output_values, 1));
-    OPENVINO_SUPPRESS_DEPRECATED_END
-    const auto autob = get_auto_broadcast();
-
-    auto out_shape = shape_infer(this,
-                                 std::vector<PartialShape>{input_values[0]->get_partial_shape(),
-                                                           input_values[1]->get_partial_shape(),
-                                                           input_values[2]->get_partial_shape()})[0]
-                         .to_shape();
-
-    output_values[0]->set_shape(out_shape);
-
-    return detail::evaluate_select(output_values, input_values, autob, output_values[0]->get_element_type());
+    OPENVINO_ASSERT(outputs.size() == 1);
+
+    const auto output_shape = shape_infer(this, ov::util::get_tensors_partial_shapes(inputs)).front().to_shape();
+    auto& output = outputs[0];
+    output.set_shape(output_shape);
+
+    const auto& cond_input = inputs[0];
+    const auto& then_input = inputs[1];
+    const auto& else_input = inputs[2];
+
+    using namespace ov::element;
+    return IfTypeOf<boolean, bf16, f16, f32, f64, i8, i16, i32, i64, u8, u16, u32, u64>::apply<select::Evaluate>(
```

Comment:
Please, port it to `IF_TYPE_OF` macro to restore Conditional Compilation support

Reply:
Done